docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,297 @@
+
+To generate factory data partition using [python script](../../scripts/tools/nrfconnect/nrfconnect_generate_partition.py) use the following command pattern:
+
+```
+$ python nrfconnect_generate_partition.py -i <path_to_JSON_file> -o <path_to_output> --offset <partition_address_in_memory> --size <partition_size> 
+```
+In this command:
+
+- <path_to_JSON_file> is a path to JSON file containing appropriate factory data.
+- <path_to_output> is a path to an output file without any prefix. E.g. By providing here */build/output* as an argument, as a result */build/output.hex* and */build/output.bin* files will be created.
+- <partition_address_in_memory> is an address in the device's persistent storage area where a partition data set should be stored.
+- <partition_size> is a size of partition in the device's persistent storage area. This value will be checked according to JSON data to check if new data fits this size.
+
+To see optional arguments for the script use the following command:
+``` 
+$ python nrfconnect_generate_partition.py -h
+```
+
+**Example of command for nRF52840DK**:
+```
+$ python nrfconnect_generate_partition.py -i build/zephyr/factory_data.json -o build/zephyr/factory_data --offset 0xfb000 --size 0x1000
+```
+
+As a result a *factory_data.hex* and *factory_data.bin* files will be created in */build/zephyr/* directory. The first contains memory offset; therefore, it can be programmed directly to the device using a programmer (E.g. nrfjprog).
+
+
+<hr>
+<a name="Building an example with factory data"></a>
+
+## Building an example with factory data
+
+Generating factory data set can be done manually using instructions described in [Generating factory data](#generating-factory-data) topic. But there is another way to generate factory data. Inside the nRF Connect build system, there is a script that generates factory data automatically using kConfigs and merges it to the firmware .hex file.
+
+To enable generating factory data set automatically go to example's directory and build an example with the following option:
+
+```
+$ west build -b nrf52840dk_nrf52840 -- -DCONFIG_CHIP_FACTORY_DATA_BUILD=y
+```
+or add `CONFIG_CHIP_FACTORY_DATA_BUILD=y` config to a ***prj.conf*** file.
+
+Each factory data parameter has default value and they are described in [Kconfig file](../../config/nrfconnect/chip-module/Kconfig). Setting a new value for factory data parameter can be done in two ways - by providing it as a build argument list or by using interactive Kconfig interfaces.
+
+### Providing factory data parameters as a build argument list
+
+This solution of providing factory data can be used with third-party build script because it uses only one command.
+All parameters can be edited manually by providing them as an additional option for the west command e.g.:
+
+```
+$ west build -b nrf52840dk_nrf52840 -- --DCONFIG_CHIP_FACTORY_DATA_BUILD=y --DCONFIG_CHIP_DEVICE_DISCRIMINATOR=0xF11
```

Comment:
Just a thought now.. do we need "_BUILD" in the KConfig name?

Reply:
I requested it to distinguish it from `FACTORY_DATA_PROVIDER`, but if it causes confusion rather than helps, we can remove it :)

Reply:
We need it because we have two separate configs for factory data. The first one is called CHIP_FACTORY_DATA_BUILD, and it enables generating of factory data during the building. The second one is CHIP_FACTORY_DATA_ENABLED, which enables factory data support. Even if the second one will be used frequently, the first of them probably will not be used by most customers. I wanted to point out that this config is used for the building only.

Reply:
Make sense to have additional suffix - just a word "BUILD" was not obvious for me that this generated the new image. But i guess it is well documented in the KConfig.